pkg/util/pki/csr.go

```diff
@@ -168,7 +168,7 @@ func BuildCertManagerKeyUsages(ku x509.KeyUsage, eku []x509.ExtKeyUsage) []v1.Ke
 // by issuers that utilise CSRs to obtain Certificates.
 // The CSR will not be signed, and should be passed to either EncodeCSR or
 // to the x509.CreateCertificateRequest function.
-func GenerateCSR(crt *v1.Certificate) (*x509.CertificateRequest, error) {
+func GenerateCSR(crt *v1.Certificate, defaultEncodeBasicConstraintsInRequest bool) (*x509.CertificateRequest, error) {
```

Comment:
suggestion: Could we avoid breaking people here? I'd also generally prefer to avoid boolean parameters like this because it's very unclear when the function is called what the parameter does.

Maybe something like:

```go
type GenerateCSROptions struct {
    DefaultEncodeBasicConstraintsInRequest bool
}

func GenerateCSR(crt *v1.Certificate) (*x509.CertificateRequest, error) {
    return GenerateCSRWithOptions(crt, GenerateCSROptions{
        DefaultEncodeBasicConstraintsInRequest: false,
    })
}

func GenerateCSRWithOptions(crt *v1.Certificate, opts GenerateCSROptions) (*x509.CertificateRequest, error) {
...
}
```

This would also avoid a lot of the diffs in this PR which look like this:

```diff
-	csr, err := utilpki.GenerateCSR(crt)
+	csr, err := utilpki.GenerateCSR(crt, true)
```

Reply:
we could also have the following signature:
```
func GenerateCSRWithOptions(crt *v1.Certificate, ...opts GenerateCSROption) (*x509.CertificateRequest, error) {
...
}
```

Reply:
Yep that works too!